client/src/search-utils.ts

```diff
@@ -32,3 +32,77 @@ export function useFocusViaKeyboard(
     };
   }, [inputRef]);
 }
+
+const COLLECTION_ITEMS_KEY: string = "collection-items";
+const COLLECTION_ITEMS_UPDATED_DATE_KEY: string =
+  "collection-items-updated-date";
+
+type Item = {
+  id: number;
+  title: string;
+  url: string;
+};
+
+function getCollectionItemsUpdatedDate(): Date | null {
+  let dateString = window?.localStorage?.getItem(
+    COLLECTION_ITEMS_UPDATED_DATE_KEY
+  );
+  if (!dateString) {
+    return null;
+  }
+  return new Date(dateString);
+}
+
+function setCollectionItemsUpdatedDate(updated: Date) {
+  window?.localStorage?.setItem(
+    COLLECTION_ITEMS_UPDATED_DATE_KEY,
+    updated.toISOString()
+  );
+}
+
+export function getCollectionItems(): Item[] {
+  return JSON.parse(
+    window?.localStorage?.getItem(COLLECTION_ITEMS_KEY) || "[]"
+  );
+}
+
+function setCollectionItems(items: Item[]) {
+  window?.localStorage?.setItem(COLLECTION_ITEMS_KEY, JSON.stringify(items));
+  window?.mdnWorker?.mutationCounter ?? window.mdnWorker.mutationCounter++;
+}
+
+export async function fetchAllCollectionsItems(remote_updated: null | Date) {
+  const local_updated = getCollectionItemsUpdatedDate();
+  if (!local_updated || !remote_updated || remote_updated > local_updated) {
```

Comment:
I think we mostly use camelCase, not snake case:

```suggestion
export async function fetchAllCollectionsItems(remoteUpdated: null | Date) {
  const localUpdated = getCollectionItemsUpdatedDate();
  if (!localUpdated || !remoteUpdated || remoteUpdated > localUpdated) {
```